distributed/scheduler.py

```diff
@@ -4138,6 +4193,52 @@ def decide_worker(self, ts):
 
         return worker
 
+    def transition_waiting_speculative(self, key):
+        try:
+
+            ts = self.tasks[key]
+
+            if self.validate:
+                # All dependencies are on the same worker
+                assert len({dts.processing_on for dts in ts.dependencies}) == 1
+
+            ws = list(ts.dependencies)[0].processing_on
+
+            worker = ws.address
```

Comment:
I have a slight preference to use `ws.address` inline.  I find that having few root terms reduces mental indirection when reviewing / reading later.  "Wait, what is worker?  Was this a parameter that was passed in?  Oh no, I see, it's just the address of the workers state."